src/screens/Assistant_v2/use-trips-query.ts

```diff
@@ -109,12 +112,16 @@ export default function useTripsQuery(
                 tripSearchPreferences,
               );
 
-              allTripPatterns = allTripPatterns.concat(
+              const tripPatternsWithKeys = decorateTripPatternWithKey(
                 results.trip.tripPatterns,
               );
 
-              setTripPatterns(allTripPatterns);
-              tripsFoundCount += results.trip.tripPatterns.length;
+              allTripPatterns = allTripPatterns.concat(tripPatternsWithKeys);
+              const filteredTripPatterns =
+                filterDuplicateTripPatterns(allTripPatterns);
+              setTripPatterns(filteredTripPatterns);
+
+              tripsFoundCount += tripPatternsWithKeys.length;
```

Comment:
Dette blir ikke helt presist siden noen av resultatene kan være filtrert ut. Men det må nok gjøres litt om på filtrerings-logikken for å få fikset dette, så du får vurdere om det er verdt innsatsen. Det er ikke kritisk i hvert fall.

Reply:
du har helt rett!
fikser.

Reply:
testet ny logikk også for ankomst søk med masse duplicates